Libraries/Azure.psm1

```diff
@@ -363,33 +363,38 @@ Function Create-AllResourceGroupDeployments($SetupTypeData, $TestCaseData, $Dist
                             }
                         }
                         else {
-                            Write-LogErr "Unable to Deploy one or more VM's"
+                            $Error = "Unable to Deploy one or more VM's. " + $CreateRGDeployments.Error
+                            Write-LogErr $Error
                             $retryDeployment = $retryDeployment + 1
                             $retValue = "False"
                             $isServiceDeployed = "False"
                         }
                     }
                     else {
-                        Write-LogErr "Unable to create $groupName"
+                        $Error = "Unable to create $groupName"
+                        Write-LogErr $Error
                         $retryDeployment = $retryDeployment + 1
                         $retValue = "False"
                         $isServiceDeployed = "False"
                     }
                 } else {
-                    Write-LogErr "Unable to delete existing resource group - $groupName"
+                    $Error = "Unable to delete existing resource group - $groupName."
+                    Write-LogErr $Error
                     $retryDeployment = 3
                     $retValue = "False"
                     $isServiceDeployed = "False"
                 }
             }
         } else {
-            Write-LogErr "Core quota is not sufficient. Stopping VM deployment."
+            $Error = "Core quota is not sufficient. Stopping VM deployment."
+            Write-LogErr $Error
             $retValue = "False"
             $isServiceDeployed = "False"
         }
     }
-    return $retValue, $deployedGroups, $resourceGroupCount, $DeploymentElapsedTime
+    return $retValue, $deployedGroups, $resourceGroupCount, $DeploymentElapsedTime, $Error
```

Comment:
if all go through true block, this variable '$Error' needs to be initialized.

Reply:
done